### DISCUSSION TITLE
Is there a way to reorder Dataset dimensions to match the DataArrays?

### DISCUSSION THREAD

**Discussion by ahuang11** [pydata/xarray#4649] [Q&A] [answered]
```
import xarray as xr
ds = xr.tutorial.open_dataset('air_temperature')
ds['airx2'] = ds['air'] * 2
ds = ds.transpose('time', 'lon', 'lat')
print(ds.dims)
```
yields
`Frozen(SortedKeysDict({'lat': 25, 'time': 2920, 'lon': 53}))`

Since it's labeled SortedKeysDict, I expected the same ordering as shown, but when I call
`list(ds.dims)`

I get:
`['lat', 'lon', 'time']`

which is different from the provided, transposed order
`('time', 'lon', 'lat')`

If I subset a DataArray, it's correct:
`print(ds['air'].dims)`
`('time', 'lon', 'lat')`

**Comment by keewis** [comment, marked as answer]
sorry for the late reply, @ahuang11.

To me the the name `SortedKeysDict` implies that it is sorted, not ordered (the difference being that – similar to a `OrderedDict` – "ordered" follows the order of insertion while "sorted" is sorted somehow, in this case alphabetically).

The difficulty with `Dataset` dimensions is that the order has no actual meaning: consider a `Dataset` with three variables, one with the dimensions `("x", "y")`, the second `("x", "t")` and the other `("z", "t")`. Which order should we use for those?

In order to completely avoid that somewhat complex decision, the `Dataset` dimensions are always sorted alphabetically, while for `DataArray` (or the semi-internal `Variable`) the order does have an actual meaning so those objects keep that. I know that for single variable `Dataset` objects that is confusing, but for multi-variable datasets this is a lot easier to maintain (and to understand).
